src/effect-store.ts

```diff
@@ -7,16 +7,17 @@
  */
 
 import { buildNamespace } from './namespace';
-import { Effect } from './types';
+import { EffectStore } from './types';
 
-const defaultEffectStore = {} as {
-	[callLikeKey: string]: Effect;
+const newDefaultEffectStore = () => {
+	return {} as EffectStore;
 };
+
```

Comment:
this is the fix - by making this a function that generates an object, rather than defaulted as an object, we ensure that new objects are created when we create the effect queue and store.